src/test/java/seedu/address/logic/parser/SettleCommandParserTest.java

```diff
@@ -0,0 +1,89 @@
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static seedu.address.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.commands.CommandTestUtil.INVALID_AMOUNT_DESC;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_AMOUNT_DESC;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_AMOUNT;
+import static seedu.address.testutil.TypicalIndexes.INDEX_FIRST_STUDENT;
+import static seedu.address.testutil.TypicalIndexes.INDEX_SECOND_STUDENT;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.SettleCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+public class SettleCommandParserTest {
+
+    private static final String MESSAGE_INVALID_FORMAT =
+            String.format(MESSAGE_INVALID_COMMAND_FORMAT, SettleCommand.MESSAGE_USAGE);
+    private SettleCommandParser parser = new SettleCommandParser();
+
+    @Test
+    public void parse_missingParts_failure() {
+        // no index specified
+        assertParseFailure(parser, VALID_AMOUNT_DESC, MESSAGE_INVALID_FORMAT);
+
+        // no amount specified
+        assertParseFailure(parser, "1",
+                String.format(MESSAGE_INVALID_COMMAND_FORMAT, SettleCommand.MESSAGE_USAGE));
+
+        // no index and no amount specified
+        assertParseFailure(parser, "", MESSAGE_INVALID_FORMAT);
+    }
+
+    @Test
+    public void parse_invalidPreamble_failure() {
+        // negative index
+        assertParseFailure(parser, "-5" + VALID_AMOUNT_DESC, MESSAGE_INVALID_FORMAT);
+
+        // zero index
+        assertParseFailure(parser, "0" + VALID_AMOUNT_DESC, MESSAGE_INVALID_FORMAT);
+
+        // invalid arguments being parsed as preamble
+        assertParseFailure(parser, "1 some random string", MESSAGE_INVALID_FORMAT);
+
+        // invalid prefix being parsed as preamble
+        assertParseFailure(parser, "1 i/ string", MESSAGE_INVALID_FORMAT);
+    }
+
+    @Test
+    public void parse_invalidAmount_failure() {
+        assertParseFailure(parser, "1" + INVALID_AMOUNT_DESC, "Amount has to positive");
+    }
+
+    @Test
+    public void parse_allFieldsSpecified_success() {
+        double amount = 100.0; // Example amount
+        String userInput = INDEX_SECOND_STUDENT.getOneBased() + " " + PREFIX_AMOUNT + amount;
+
+        SettleCommand expectedCommand = new SettleCommand(INDEX_SECOND_STUDENT, amount);
+
+        assertParseSuccess(parser, userInput, expectedCommand);
+    }
+
+    @Test
+    public void parse_multipleRepeatedFields_failure() {
+        String userInput = INDEX_FIRST_STUDENT.getOneBased() + VALID_AMOUNT_DESC + INVALID_AMOUNT_DESC;
+
+        assertParseFailure(parser, userInput, Messages.getErrorMessageForDuplicatePrefixes(PREFIX_AMOUNT));
+    }
+
+    // Helper methods for assertions
+
+    private void assertParseFailure(SettleCommandParser parser, String userInput, String expectedMessage) {
```

Comment:
Hmm, isn't this already defined in `CommandParserTestUtil.java`?

Reply:
Thank you for pointing that out. I have made changes accordingly.